spec/integration/util/windows/user_spec.rb

```diff
@@ -117,6 +117,28 @@ def expect_logon_failure_error(&block)
       it "should return false given a nil username and an incorrect password" do
         expect(Puppet::Util::Windows::User.password_is?(nil, bad_password)).to be_falsey
       end
+
+      context "with a correct password" do
+        it "should return true even if account restrictions are in place " do
+          Puppet::Util::Windows::User.stubs(:logon_user).raises(Puppet::Util::Windows::Error.new('', 1327))
```

Comment:
Do we want to add any negative tests here for the other error codes?

```
ERROR_PASSWORD_MUST_CHANGE = 1907
ERROR_LOGON_FAILURE = 1326
ERROR_NO_LOGON_SERVERS = 1311
ERROR_ACCOUNT_LOCKED_OUT = 1909
ERROR_ACCOUNT_EXPIRED = 1793
ERROR_PASSWORD_EXPIRED = 1330
```

`1909` in particular is an interesting one... because from what I've read, that one can be raised even when the creds are valid.